### PR TITLE
feat(feedback): button to hide/show user feedback in issue details

### PR DESCRIPTION
**Pull request by aliu39**
Depends on https://github.com/getsentry/sentry/pull/74228

Relates to https://github.com/getsentry/sentry/issues/74209

Adds a button to the "User Feedback" section of issue details page. Allows members to hide/unhide feedback from **all** issue details in a project. Persists this decision across reloads using the prompt backend

Updated demo with chevron/drop-down style hiding:

https://github.com/user-attachments/assets/43a02a63-98c0-42bd-8651-f1baeeb22218

Updated loading indicator: (you can see for a split sec at :12)

https://github.com/user-attachments/assets/a2d57c51-e459-467b-9a9c-5e227050d9b7

### CODE REVIEW

**Comment by aliu39** [comment]
This component, EventUserFeedback, is used in other places too. For ex the User Feedback tab of issues. So we'll have to move the usePrompt and isPromptDismissed check to one of its parents, probably `EventDataSection`.

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 1]
```diff
@@ -131,13 +149,31 @@ function DefaultGroupEventDetailsContent({
...
+                  isHidden={feedbackHidden}
+                  showFeedback={showPrompt}
+                  hideFeedback={dismissPrompt}
+                />
+              </ButtonBar>
+            </ErrorBoundary>
+          }
+        >
+          {feedbackHidden ? (
+            '(Hidden)'
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:168]: style thing but could we instead have something like `This feedback is hidden.` in grey text and italicized? i think that would match sentry style a bit more

File: static/app/actionCreators/prompts.tsx [hunk 2]
```diff
@@ -215,12 +226,39 @@ export function usePrompt({
...
+  const showPrompt = useCallback(() => {
+    promptsUpdate(api, {
+      organization,
+      projectId,
+      feature,
+      status: 'visible',
+    });
+
+    // Update cached query data
+    // Will set prompt to snoozed
```
Comment (michellewzhang) [inline static/app/actionCreators/prompts.tsx:238]: why are we setting to snoozed?

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 3]
```diff
@@ -131,13 +149,31 @@ function DefaultGroupEventDetailsContent({
...
-            orgSlug={organization.slug}
-            issueId={group.id}
-            showEventLink={false}
-          />
+        <EventDataSection
+          title={t('User Feedback')}
+          type="user-feedback"
+          actions={
+            <ErrorBoundary mini>
+              <ButtonBar gap={1}>
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:157]: we don't need a `ButtonBar` here if it's just 1 button

File: static/app/actionCreators/prompts.tsx (hunk 2 above)
Comment (aliu39) [reply static/app/actionCreators/prompts.tsx:227]: Forgot to update comment

File: static/app/components/events/userFeedback.tsx [hunk 4]
```diff
@@ -11,6 +11,36 @@ import type {UserReport} from 'sentry/types/group';
 import {escape, nl2br} from 'sentry/utils';
 import useCopyToClipboard from 'sentry/utils/useCopyToClipboard';
 
+// used for issue details page (groupEventDetailsContent.tsx)
+export function HideUserFeedbackButton({
```
Comment (aliu39) [inline static/app/components/events/userFeedback.tsx:15]: I'm thinking of moving this to groupEventDetailsContent.tsx

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 5]
```diff
@@ -131,13 +148,33 @@ function DefaultGroupEventDetailsContent({
...
+                <HideUserFeedbackButton
+                  isLoading={promptLoading}
+                  isError={promptError || isPromptDismissed === undefined}
+                  isHidden={isPromptDismissed !== undefined ? isPromptDismissed : false}
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:160]:
```suggestion
                  isHidden={isPromptDismissed ?? false}
```

first part will automatically check for `undefined`

File: static/app/components/events/userFeedback.tsx (hunk 4 above)
Comment (michellewzhang) [reply static/app/components/events/userFeedback.tsx:15]: might be good, could reduce a bunch of prop passing

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 6]
```diff
@@ -131,13 +148,33 @@ function DefaultGroupEventDetailsContent({
...
+                  showFeedback={showPrompt}
+                  hideFeedback={dismissPrompt}
+                />
+              </ButtonBar>
+            </ErrorBoundary>
+          }
+        >
+          {promptLoading || promptError || isPromptDismissed ? (
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:168]: should we have some different states for loading & error? could use `<LoadingIndicator>` or `<LoadingError>`

Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:195]: I'm thinking of defaulting to displaying the feedback in either of these cases. So if the prompt API is having problems, users will still be able to see their feedback.

Comment (michellewzhang) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:195]: that makes sense to me

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 7]
```diff
@@ -131,13 +177,33 @@ function DefaultGroupEventDetailsContent({
...
+          actions={
+            // Button is disabled if promptLoading or promptError.
+            <ErrorBoundary mini>
+              <HideUserFeedbackButton
+                isLoading={promptLoading}
+                isError={promptError || isPromptDismissed === undefined}
+                isHidden={isPromptDismissed ?? false}
+                showFeedback={showPrompt}
+                hideFeedback={dismissPrompt}
+              />
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:192]: would it be better to just move the button code here directly? it's not that much more code, and you'll have less prop passing

Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:192]: Truu

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 8]
```diff
@@ -131,13 +148,38 @@ function DefaultGroupEventDetailsContent({
...
+              <Button
+                size="xs"
+                onClick={isPromptDismissed ? showPrompt : dismissPrompt}
+                title={
+                  isPromptDismissed
+                    ? t('Unhide feedback on all issue details')
+                    : t('Hide feedback on all issue details')
+                }
+                disabled={promptError || promptLoading || isPromptDismissed === undefined}
+              >
```
Comment (michellewzhang) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: what does it mean if `isPromptDismissed` undefined?

Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: https://github.com/getsentry/sentry/blob/138a24e16e507fdb66aa231a0f4300bcb37d62ce/static/app/actionCreators/prompts.tsx#L162 I think some kind of error with the backend. I'm assuming when this occurs promptError=true.

Comment (michellewzhang) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: i think it shouldn't be disabled if `isPromptDismissed === undefined` -- it looks like it's undefined anytime the prompt is not dismissed, even with no error


SCR-20240715-lqtw

Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: oh interesting.. thanks for testing this case!

Comment (michellewzhang) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: actually, hold on... `isPromptDismissed` is supposed to return `true` or `false` so the logic isn't adding up. maybe it's a vercel preview thing. does https://sentry-sdks.sentry.io/issues/5291936195/events/e24fd30c0f2b47a9b20bc5014de8e493/?project=-1&referrer=event-or-group-header&statsPeriod=90d work as expected for you?

Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: Using the latest commit, the button is disabled for me

**Comment by michellewzhang** [comment]
might need a loading state here 

https://github.com/user-attachments/assets/68beca83-b09d-4f32-9176-5038628a0baf

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx (hunk 8 above)
Comment (aliu39) [reply static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:166]: Agreed on slack to show the feedback if error or `isPromptDismissed === undefined`, use `LoadingIndicator` when loading

**Review by michellewzhang — APPROVED** [review]
oo i like the new loading indicator! i know @billyvg said it adds noise to the page, so we could always do `Placeholder` instead but it definitely looks a lot better. cc @Jesse-Box for design thoughts

**Pull request merged**